### PR TITLE
Update psautohint to use autohint CLI options

### LINKED ISSUE #129: CLI: `-o` parameter order dependant
psautohint errors out when I invoke it like `psautohint -o output.otf input.otf` but works fine if I do `psautohint input.otf -o output.otf`.

```
> .\venv\Scripts\psautohint -o .\build\desktop\asdf_Bd.otf .\build\unhinted\asdf_Bd.otf
usage: psautohint [-h] [-v] [-o PATH [PATH ...]] [-r PATH] [-a] [-w]
                  [-g GLYPH_LIST | --glyphs-file PATH | -x GLYPH_LIST | --exclude-glyphs-file PATH]
                  [-c | --report-only] [--log PATH] [-d] [--no-flex]
                  [--no-hint-sub] [--no-zones-stems] [--fontinfo-file PATH]
                  [--print-dflt-fddict] [--print-list-fddict] [--doc-fddict]
                  [--info] [--version]
                  FONT [FONT ...]
psautohint: error: the following arguments are required: FONT
```

Edit: just saw that `-o` is greedy. Hm...

### LINKED ISSUE #176: Add arg flags compatible with afdko autohint
Update `argparse` configuration in `psautohint` to accept the same single-hyphen flags as afdko's autohint program for the same flag meaning. For example, add `-xgf` for `--exclude-glyphs-file`.

### PR DESCRIPTION
**Pull request by josh-hadley**
Use legacy `autohint` CLI flags & fix "intuitive" use of `-o` option (output preceding input).

closes #176 
closes #129

### CODE REVIEW

File: python/psautohint/__main__.py [hunk 1]
```diff
@@ -385,6 +386,35 @@ def __call__(self, parser, namespace, values, option_string=None):
         parser.exit(message=formatter.format_help())
 
 
+class _DeprecatedAction(argparse.Action):
+    """
+    Custom action to raise a DeprecationWarning when hit. Used for the
+    autohint compatibility args/flags.
+    """
+    def __init__(self,
```
Comment (miguelsousa) [inline python/psautohint/__main__.py]: add blank line above `def __init__` (and also at line no. 369)

File: python/psautohint/__main__.py [hunk 2]
```diff
@@ -691,6 +846,18 @@ def get_options(args):
     for handler in logging.root.handlers:
         handler.addFilter(DuplicateMessageFilter())
 
+    if (not len(parsed_args.font_paths or []) and
+            len(parsed_args.output_paths or [])):
+        # allow "psautohint -o outputpath inputpath"
+        # see https://github.com/adobe-type-tools/psautohint/issues/129
+        half = len(parsed_args.output_paths)//2
```
Comment (miguelsousa) [inline python/psautohint/__main__.py]: add spaces around `//`

File: python/psautohint/__main__.py (hunk 1 above)
Comment (josh-hadley) [reply python/psautohint/__main__.py]: done

File: python/psautohint/__main__.py (hunk 2 above)
Comment (josh-hadley) [reply python/psautohint/__main__.py]: done

**Comment by josh-hadley** [comment]
Closing, needs some additional work. I found some additional issues with arg ordering for the autohint-compat args that aren't picked up in testing.

**Comment by josh-hadley** [comment]
@miguelsousa @khaledhosny : re-opened with fixes. This PR is once again ready to be reviewed, when you get a chance.

**Comment by miguelsousa** [comment]
@khaledhosny please provide your feedback on these changes when you have a chance. Thanks.

File: tests/integration/test_cli.py [hunk 3]
```diff
@@ -295,3 +295,67 @@ def test_stemhist(args, tmpdir):
             g = args[args.index('-g') + 1]
             exp_suffix = '.' + g + exp_suffix
         assert differ([path + exp_suffix, out + suffix, '-l', '1'])
+
+
+@pytest.mark.parametrize("path", FONTS)
+def test_outpath_intuitive(path, tmpdir):
```
Comment (khaledhosny) [inline tests/integration/test_cli.py]:
```suggestion
def test_outpath_order(path, tmpdir):
```

Comment (josh-hadley) [reply tests/integration/test_cli.py]: Changed this and a couple of others (replacing "intuitive" with "order").

**Pull request merged**